lua/csharp/features/user-secrets/init.lua

```diff
@@ -0,0 +1,95 @@
+local M = {}
+local logger = require("csharp.log")
+local utils = require("csharp.utils")
+local dotnet_cli = require("csharp.modules.dotnet-cli")
+
+-- Creates the user secret file if the same
+-- doesn't exists
+local function _ensure_secret_exists(user_secret_file_path)
+  local file, _ = io.open(user_secret_file_path, "r+")
+
+  if not file then
+    os.execute("mkdir -p " .. user_secret_file_path)
+    os.execute("chmod 600 " .. user_secret_file_path)
+  end
+end
+
+-- Paramter:
+-- • {output_init_command} Output from init command
+--
+-- Returns a string with the user id or nil.
+--- @param output_init_command string
+--- @return string | nil
+local function _extract_user_id_from_command(output_init_command)
+  return string.match(output_init_command, "Set UserSecretsId to '(.*)'")
+end
+
+local function _extract_user_id_from_project(project_path)
+  local project_file = io.open(project_path, "r")
+  local user_id
+
+  if not project_file then
+    logger.error("Something went wrong during read the project file", { feature = "user-secrets" })
+    return
+  end
+
+  for line in project_file:lines() do
+    user_id = string.match(line, "<UserSecretsId>(.*)</UserSecretsId>")
+    if user_id then
+      break;
+    end
+  end
+
+  project_file:close()
+  return tostring(user_id)
+end
+
+--
+-- Paramter:
+-- • {project_path} Path to project.
+--- @param project_path string
+--- @return string | nil
+local function _init_secret(project_path)
+  local output, _ = dotnet_cli.user_secrets("init", project_path)
+
+  if output then
+    return _extract_user_id_from_command(output)
+  end
+end
+
+local function _open_secret_in_buffer(user_secret_id)
+  local secret_file_path = os.getenv("HOME") .. "/.microsoft/usersecrets/" .. user_secret_id .. "/secrets.json"
+  _ensure_secret_exists(secret_file_path)
+
+  vim.cmd.edit(secret_file_path)
```

Comment:
I suggest we create an empty json object `{}` after we create the user-secrets file. Because saving an empty file may result in unexpected behavior.

Reply:
Done! Good point by the way! Thanks